### PR TITLE
[Communication] 0.1.0 Init

### PR DESCRIPTION
**Pull request by haroldrandom**
Close https://github.com/Azure/azure-cli-pr/issues/99

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Communication

File: src/communication/azext_communication/generated/_help.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+# pylint: disable=too-many-lines
+
+from knack.help_files import helps
+
+
+helps['communication service'] = """
```
Comment (Juliehzl) [inline src/communication/azext_communication/generated/_help.py:15]: It seems that there are only two command groups in `az communication`. If yes, why not move all commands in `service` command group to `communication` level?

File: src/communication/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+This is the extension for communication
+
+### How to use ###
+Install this extension using the below CLI command
+```
+az extension add --name communication
+```
+
+### Included Features ###
+#### communication communication-service ####
```
Comment (Juliehzl) [inline src/communication/README.md:11]: Not compatible with command group description.
aligning with `az communication service` will be better.

File: src/communication/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+##### Create #####
+```
+az communication communication-service create --name "MyCommunicationResource" --location "Global" \
```
Comment (Juliehzl) [inline src/communication/README.md:14]: `az communication service` should be the right command group?

File: src/communication/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    --data-location "United States" --resource-group "MyResourceGroup" 
+
+az communication communication-service wait --created --name "{myCommunicationService}" --resource-group "{rg}"
```
Comment (Juliehzl) [inline src/communication/README.md:17]: If  you want to show `wait` command usage here, please add `--no wait` in `create` command.

File: src/communication/azext_communication/generated/_help.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    type: group
+    short-summary: communication service
```
Comment (Juliehzl) [inline src/communication/azext_communication/generated/_help.py:17]: Prefer refining the help in manual

File: src/communication/azext_communication/generated/_help.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+"""
+
+helps['communication service list'] = """
+    type: command
+    short-summary: "Handles requests to list all resources in a subscription."
```
Comment (Juliehzl) [inline src/communication/azext_communication/generated/_help.py:22]: we also could list resources in a resource group, so the description is not very suitable

File: src/communication/azext_communication/generated/_help.py [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+      - name: Get resource
+        text: |-
+               az communication service show --name "MyCommunicationResource" --resource-group "MyResourceGroup"
+"""
+
+helps['communication service create'] = """
+    type: command
+    short-summary: "Create a new CommunicationService or update an existing CommunicationService."
+    examples:
+      - name: Create or update resource
```
Comment (Juliehzl) [inline src/communication/azext_communication/generated/_help.py:42]: curious about why short-summary and examples description are different

File: src/communication/azext_communication/generated/_help.py [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+
+helps['communication service regenerate-key'] = """
+    type: command
+    short-summary: "Regenerate CommunicationService access key. PrimaryKey and SecondaryKey cannot be regenerated at \
+the same time."
+    examples:
+      - name: Regenerate key
+        text: |-
+               az communication service regenerate-key --name "MyCommunicationResource" --key-type "Primary" \
+--resource-group "MyResourceGroup"
```
Comment (Juliehzl) [inline src/communication/azext_communication/generated/_help.py:96]: what about adding these commands in one new command group like `az storage account keys`?

File: src/communication/azext_communication/generated/_help.py [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+    short-summary: communication status
+"""
+
+helps['communication status show'] = """
+    type: command
+    short-summary: "Gets the current status of an async operation."
+    examples:
+      - name: Get OperationStatus
+        text: |-
+               az communication status show --operation-id "db5f291f-284d-46e9-9152-d5c83f7c14b8" --location "westus2"
```
Comment (Juliehzl) [inline src/communication/azext_communication/generated/_help.py:124]: If there is only one command, no need to have the command group. consider `show-status`

File: src/communication/README.md (hunk 2 above)
Comment (haroldrandom) [reply src/communication/README.md:11]: Fixed

File: src/communication/README.md (hunk 3 above)
Comment (haroldrandom) [reply src/communication/README.md:14]: Fixed

File: src/communication/azext_communication/generated/_help.py (hunk 6 above)
Comment (haroldrandom) [reply src/communication/azext_communication/generated/_help.py:22]: Fixed

File: src/communication/azext_communication/generated/_help.py (hunk 7 above)
Comment (haroldrandom) [reply src/communication/azext_communication/generated/_help.py:42]: short summary refined

File: src/communication/azext_communication/generated/_help.py (hunk 5 above)
Comment (haroldrandom) [reply src/communication/azext_communication/generated/_help.py:17]: Done

File: src/communication/azext_communication/generated/_help.py (hunk 9 above)
Comment (haroldrandom) [reply src/communication/azext_communication/generated/_help.py]: done

File: src/communication/azext_communication/generated/_help.py (hunk 1 above)
Comment (haroldrandom) [reply src/communication/azext_communication/generated/_help.py]: done

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**